tensorflow/c/experimental/filesystem/plugins/gcs/gcs_filesystem.cc

```diff
@@ -90,8 +81,44 @@ static void ParseGCSPath(absl::string_view fname, bool object_empty_ok,
 // SECTION 1. Implementation for `TF_RandomAccessFile`
 // ----------------------------------------------------------------------------
 namespace tf_random_access_file {
+typedef struct GCSFile {
+  const char* bucket;
+  const char* object;
```

Comment:
We should make these strings to automate memory management. Since they don't get exposed across the plugin boundary this is a good change to make.

Reply:
I saw you use `const char*` in posix filesystem. Should I make a PR for posix as well ?

Reply:
You're right. I didn't convert them to `string` there to not have a conversion from `const char*` to `string` when creating the file (API uses `const char*`) and then again from `string` to `const char*` when displaying the error message. Your choice of whether to convert or not, but in this case you have 2 pointers to keep track of and both are generated by the same function call.